idaes/apps/grid_integration/tests/test_PEM_Parameterized_bidder.py

```diff
@@ -0,0 +1,198 @@
+        ValueError, match=r".*The power of PEM is greater than the renewabele power.*"
+    ):
+        PEM_bidder = PEMParametrizedBidder(
+            bidding_model_object,
+            day_ahead_horizon,
+            real_time_horizon,
+            solver,
+            forecaster,
+            renewable_mw,
+            pem_marginal_cost,
+            pem_mw,
+        )
+
+
+def wind_df():
+    """
+    This is to define a dataframe fed to the PerfectForecaster with example DA/RT
+    capacity factors and LMPs.
+    """
+    start_year = 2020
+    start_mon = 1
+    start_day = 1
+    start_date = pd.Timestamp(f"{start_year}-{start_mon:02d}-{start_day:02d} 00:00:00")
+    ix = pd.date_range(
+        start=start_date,
+        end=start_date
+        + pd.offsets.DateOffset(days=1)
+        - pd.offsets.DateOffset(hours=24 - day_ahead_horizon + 1),
+        freq="1H",
+    )
+    df = pd.DataFrame(index=ix)
+    gen = testing_renewable_data.gen_name
+    bus = testing_renewable_data.bus
+    df[f"{gen}-DACF"] = [0.1, 0.1, 0.1, 0.8, 0.8, 0.8]
+    df[f"{gen}-RTCF"] = [1.0, 0.2, 0.2, 0.75, 0.75, 0.75]
+    df[f"{bus}-DALMP"] = [10] * day_ahead_horizon
+    df[f"{bus}-RTLMP"] = [20] * real_time_horizon
+    return df
+
+
+@pytest.fixture
+def bidder_object():
+    """
+    This is to define a bidder object.
+    """
+    example_wind_df = wind_df()
+    forecaster = PerfectForecaster(example_wind_df)
+    bidding_model_object = ExampleModel(model_data=testing_renewable_data)
+    bidder_object = PEMParametrizedBidder(
+        bidding_model_object=bidding_model_object,
+        day_ahead_horizon=day_ahead_horizon,
+        real_time_horizon=real_time_horizon,
+        solver=solver,
+        forecaster=forecaster,
+        renewable_mw=200,
+        pem_marginal_cost=30,
+        pem_mw=100,
+    )
+    return bidder_object
+
+
+@pytest.mark.component
+def test_compute_DA_bids(bidder_object):
+    """
+    This is to test the if the compute_DA_bids function works correctly.
+    """
+    gen = bidder_object.generator
+    pmin = bidder_object.bidding_model_object.pmin
+    pmax = bidder_object.bidding_model_object.pmax
+    pem_pmax = bidder_object.pem_mw
+    pem_marginal_cost = bidder_object.pem_marginal_cost
+    date = "2020-01-01"
+
+    # test DA bidding
+    bids = bidder_object.compute_day_ahead_bids(date=date, hour=0)
+    expected_DA_cf = [0.1, 0.1, 0.1, 0.8, 0.8, 0.8]
+    expected_bids = {}
+    for t in range(day_ahead_horizon):
+        expect_da_wind = expected_DA_cf[t] * pmax
+        if t <= 2:
+            expect_bids_curve = [(0, 0), (expected_DA_cf[t] * pmax, pem_marginal_cost)]
+        else:
+            expect_bids_curve = [
+                (0, 0),
+                (expected_DA_cf[t] * pmax - pem_pmax, 0),
+                (expected_DA_cf[t] * pmax, pem_marginal_cost),
+            ]
+        expect_cost_curve = convert_marginal_costs_to_actual_costs(expect_bids_curve)
+        expected_bids[t] = {

```

Comment:
This test is re-implementing the code/logic of the bidder. While this is possible because the logic is straightforward and simple, it seems to me like the test should use actual numbers for the expected bid curves so that the test pass/fail is not implementation-specific

Reply:
Was this comment addressed? If so, reply here.

Reply:
@Xinhe-Chen Please reply here if you addressed this.

Reply:
Yes this has been addressed by me.